torch_geometric/nn/conv/gcn_conv.py

```diff
@@ -61,7 +62,8 @@ def gcn_norm(edge_index, edge_weight=None, num_nodes=None, improved=False,
             edge_weight = tmp_edge_weight
 
         row, col = edge_index[0], edge_index[1]
-        deg = scatter_add(edge_weight, col, dim=0, dim_size=num_nodes)
+        idx = col if flow == "source_to_target" else row
```

Comment:
We need to do something similar if `edge_index` is a sparse tensor, probably to `dim` variable in `sparsesum`.

Reply:
I'm not 100% sure about PyG internal workings, but from what I understand, when `edge_index` is a sparse tensor, only `flow="source_to_target"` is allowed (it will raise error otherwise), even though technically it is doing aggregation from target to source (assuming first dim of `edge_index` is source, second dim is target). This is also reflected in the documentation (see [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/sparse_tensor.html), "transpose sparse matrix") and unit tests.
Therefore, technically `flow="target_to_source"` is not possible in `gcn_norm()` function, and we don't need to handle that for sparse tensor. Always calculating the degree by summing across `dim=1` should always be correct. Changing the flow for sparse tensor should be by transposing the matrix I believe, not by changing the `flow` argument.

Reply:
Ah thanks for pointing that out.